frappe/integrations/doctype/ldap_settings/test_ldap_settings.py

```diff
@@ -1,8 +1,1071 @@
+				search_base=self.ldap_user_path,
+				search_filter=self.TEST_LDAP_SEARCH_STRING.format("posix.user"),
+				attributes=self.test_class.get_ldap_attributes())
+
+		test_ldap_entry = self.connection.entries[0]
+
+		method_return = self.test_class.convert_ldap_entry_to_dict(test_ldap_entry)
+
+		self.assertTrue(type(method_return) == dict) # must be dict
+		self.assertTrue(len(method_return) == 6) # there are 6 fields in mock_ldap for use
+
+
+
+
+class Test_OpenLDAP(LDAP_TestCase, unittest.TestCase):
+	TEST_LDAP_SERVER = 'OpenLDAP'
+	TEST_LDAP_SEARCH_STRING = '(uid={0})'
+	DOCUMENT_GROUP_MAPPINGS = [
+				{
+					"doctype": "LDAP Group Mapping",
+					"ldap_group": "Administrators",
+					"erpnext_role": "System Manager"
+				},
+				{
+					"doctype": "LDAP Group Mapping",
+					"ldap_group": "Users",
+					"erpnext_role": "Blogger"
+				},
+				{
+					"doctype": "LDAP Group Mapping",
+					"ldap_group": "Group3",
+					"erpnext_role": "Accounts User"
+				}
+	]
+	LDAP_USERNAME_FIELD = 'uid'
+	LDAP_SCHEMA = OFFLINE_SLAPD_2_4
+	LDAP_LDIF_JSON = 'test_data_ldif_openldap.json'
+
+	TEST_VALUES_LDAP_COMPLEX_SEARCH_STRING =  [
+			'(uid={0})',
+			'(&(objectclass=posixaccount)(uid={0}))',
+			'(&(description=*ACCESS:test1*)(uid={0}))', # OpenLDAP has no member of group, use description to filter posix.user has equivilent of AD 'memberOf'
+			'(&(objectclass=posixaccount)(description=*ACCESS:test3*)(uid={0}))' # OpenLDAP has no member of group, use description to filter posix.user doesn't have. equivilent of AD 'memberOf'
+	]
+
+
+class Test_ActiveDirectory(LDAP_TestCase, unittest.TestCase):
+	TEST_LDAP_SERVER = 'Active Directory'
+	TEST_LDAP_SEARCH_STRING = '(samaccountname={0})'
+	DOCUMENT_GROUP_MAPPINGS = [
+				{
+					"doctype": "LDAP Group Mapping",
+					"ldap_group": "Domain Administrators",
+					"erpnext_role": "System Manager"
+				},
+				{
+					"doctype": "LDAP Group Mapping",
+					"ldap_group": "Domain Users",
+					"erpnext_role": "Blogger"
+				},
+				{
+					"doctype": "LDAP Group Mapping",
+					"ldap_group": "Enterprise Administrators",
+					"erpnext_role": "Accounts User"
+				}
+	]
+	LDAP_USERNAME_FIELD = 'samaccountname'
+	LDAP_SCHEMA = OFFLINE_AD_2012_R2
+	LDAP_LDIF_JSON = 'test_data_ldif_activedirectory.json'
+
+	TEST_VALUES_LDAP_COMPLEX_SEARCH_STRING =  [
+			'(samaccountname={0})',
+			'(&(objectclass=user)(samaccountname={0}))',
+			'(&(description=*ACCESS:test1*)(samaccountname={0}))', # OpenLDAP has no member of group, use description to filter posix.user has equivilent of AD 'memberOf'
+			'(&(objectclass=user)(description=*ACCESS:test3*)(samaccountname={0}))' # OpenLDAP has no member of group, use description to filter posix.user doesn't have. equivilent of AD 'memberOf'
+	]
+
+
+
+
+
+
+
+
+
+
+##############################################################################################################
+################ ToDo: Data below this block should be removed prior to last code review #####################
```

Comment:
refactor:

Move the data below this line to a gist and place a link to it in this PR should someone wish to use it to create data.

Reply:
added to [this gist](https://gist.github.com/jon-nfc/70fda69d158d701a6fb714172408bc0b)

Reply:
removed in 9f92fa02de69039990dd3b913df6a1da41681dbd